File: 2695-find-score-of-an-array-after-marking-all-elements/2695-find-score-of-an-array-after-marking-all-elements.cs
public class Solution {
    public long FindScore(int[] nums) {
        int n = nums.Length;
        long score = 0;

        var valInd = new List<(int val, int ind)>();
        for (int i = 0; i < n; i++) {
            valInd.Add((nums[i], i));
        }

        // Sort by value, then index
        valInd.Sort((a, b) => a.val == b.val ? a.ind - b.ind : a.val - b.val);

        bool[] marked = new bool[n];

        foreach (var (val, ind) in valInd) {
            if (marked[ind]) continue;

            score += val;

            if (ind > 0) marked[ind - 1] = true;
            marked[ind] = true;
            if (ind < n - 1) marked[ind + 1] = true;
        }

        return score;
    }
}

File: 2695-find-score-of-an-array-after-marking-all-elements/2695-find-score-of-an-array-after-marking-all-elements.java
import java.util.*;

class Solution {
    public long findScore(int[] nums) {
        int n = nums.length;
        long score = 0;

        List<int[]> valInd = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            valInd.add(new int[]{nums[i], i});
        }

        // Sort by value, then by index
        Collections.sort(valInd, (a, b) -> a[0] == b[0] ? a[1] - b[1] : a[0] - b[0]);

        boolean[] marked = new boolean[n];

        for (int[] p : valInd) {
            int val = p[0];
            int ind = p[1];

            if (marked[ind]) continue;

            score += val;

            if (ind > 0) marked[ind - 1] = true;
            marked[ind] = true;
            if (ind < n - 1) marked[ind + 1] = true;
        }

        return score;
    }
}
